### PR TITLE
files: optimizes the files loading time.

### PR DESCRIPTION
**Pull request by jma**
* Updates primeng dependency.
* Creates a specific upload files components.
* Adds loading wheels.

### CODE REVIEW

File: projects/admin/src/app/record/detail-view/document-detail-view/files-collections/upload-files/upload-files.component.ts [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+
+  /**
+   * Filter the uploaded files.
+   *
+   * @param event the standard event.
+   * @param _ unused.
+   */
+  onSelect(event, _) {
+    const existingFileNames = [];
+    for (var i = 0; i < event.files.length; i++) {
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/files-collections/upload-files/upload-files.component.ts:213]: `let` ?

or use `.map` on event.files  ?

File: projects/admin/src/app/service/resources-files.service.ts [hunk 2]
```diff
@@ -126,11 +122,8 @@ export class ResourcesFilesService {
    * @param pid PID of the record.
    * @returns Observable resolving the list of files.
    */
-  list(type: string, pid: string, parentRecord: Record): Observable<Array<File>> {
-    if (parentRecord == null) {
-      return of([]);
-    }
-    return this.httpService.get(`${this.baseUrl}/${parentRecord.id}/files`).pipe(
+  list(parentRecordId: string): Observable<Array<File>> {
```
Comment (Garfield-fr) [inline projects/admin/src/app/service/resources-files.service.ts:125]: `Observable<File[]>` ?

**Review by Garfield-fr — APPROVED** [review]

File: projects/admin/src/app/record/detail-view/document-detail-view/files-collections/upload-files/upload-files.component.ts (hunk 1 above)
Comment (jma) [reply projects/admin/src/app/record/detail-view/document-detail-view/files-collections/upload-files/upload-files.component.ts:213]: unfortunately not it is not a standard array.

File: projects/admin/src/app/service/resources-files.service.ts (hunk 2 above)
Comment (jma) [reply projects/admin/src/app/service/resources-files.service.ts:125]: No. It is an array of files.

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.ts [hunk 3]
```diff
@@ -78,13 +78,21 @@ export class DocumentDetailViewComponent implements DetailRecord, OnInit, OnDest
...
+  get showLocalFieldsTab(): boolean {
     return this.permissionsService.canAccess([PERMISSIONS.LOFI_SEARCH, PERMISSIONS.LOFI_CREATE]);
   }
 
+  /**
+   * Show or hide files tab
+   * @return boolean - if False, hide the local fields tab
+   */
+  get showFilesTab(): boolean {
+    return this.permissionsService.canAccess(PERMISSIONS.CIRC_ADMIN);
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.ts:93]: A new permission for files ?

**Review by Garfield-fr — APPROVED** [review]

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.ts (hunk 3 above)
Comment (jma) [reply projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.ts:93]: No, according to the PO specification files has the same permission to the circulation.

**Review by lauren-d — APPROVED** [review]

**Pull request merged**